controllers/userController.js

```diff
@@ -1,10 +1,43 @@
+const redis = require("redis");
 const { promisify } = require("util");
 const User = require("../models/userModel");
 const catchAsync = require("../utils/catchAsync");
 const jwt = require("jsonwebtoken");
 const AppError = require("../utils/appError");
+const sendEmail = require("../utils/email");
 const { validationResult } = require("express-validator");
-const { constants } = require("../utils/constants");
+const crypto = require("crypto");
+const { constants, statusCodes } = require("../utils/constants");
+
+const client = redis.createClient();
+
+const generateOTP = () => {
+  return crypto.randomInt(100000, 999999);
+};
+
+const otpCode = generateOTP();
+client.on("error", (err) => console.log("Redis Client Error", err));
+const startRedis = async () => {
+  try {
+    await client.connect(); // Establish connection
```

Comment:
Also, to be sure, this connection is only called on startup, right?

Reply:
yes, just once